### ISSUE TITLE
Keys in manifest not actually source file names

### ISSUE THREAD

**Issue by jacobwindsor** [shellscape/webpack-manifest-plugin#110]
In the docs, it states "This will generate a manifest.json file in your root output directory with a mapping of all source file names to their corresponding output file..."

However, in actuality, the key is generated from the entry name. E.g.

```js
// webpack.config.js
...
entry: {
        'myentry': './path/to/my/entry/index.js'
    },
...
```

gives:
```
// manifest.json
{
  "myentry.js": "/dist/myentry.bundle.js",
}
```

I think this is actually better behaviour then simply using the source file name. However, the only inconsistency is the addition of the `.js` suffix in manifest.json. There should be an option to use the entry name (defined in webpack.config.js) or the source file name.

**Comment by a-x-** [comment]
It makes sense for me

Keys are filenames with an assumption that webpack `output.filename` has a default value `[name].js`

What about `entryNameKeys: Bool` option?

**Comment by a-x-** [comment]
possible code place: https://github.com/danethurber/webpack-manifest-plugin/blob/master/lib/plugin.js#L54

**Comment by a-x-** [comment]
Alternatively we can create optional function `formatKey` with some arguments like `{ entryName, ext, chunk, files }` with webpack plugin context (`call(this, ...)`)

**Comment by jacobwindsor** [comment]
The `entryNameKeys ` option seems more appropriate to me. And with what the docs say, should be true by default. If the addition of the js extension was by default (as is now) the docs should be reworded to clarify that. The entry name keys in manifest aren't actually the source file names, they are the entry names with a js extension. While a lot of people would have the entry name to be the same as the source file name, I think assuming that is dangerous.

Happy to make a PR once a decision has been made.

**Comment by a-x-** [comment]
I agree,
may be we will create major release some time with entry name keys by default

I think it's possible to add an some option for now

will see what @mastilver position

**Comment by jacobwindsor** [comment]
@a-x- @mastilver Since there was no activity on this for a while, I went ahead and created a PR hoping everyone agrees. I need this functionality for my own use case so just trying to speed things up.

Hope it's okay!

**Comment by shellscape** [comment]
Fixed in a998c4b
